### PR TITLE
bcm27xx-eeprom: update and split into bcm2711/bcm2712

### PR DESCRIPTION
**Pull request by Noltari**
Maintainer: me
Compile tested: bcm27xx/bcm2711
Run tested: bcm27xx/bcm2711

Description: Updates Raspberry Pi 4 EEPROM and adds Raspberry Pi 5 EEPROM support.
Adds separate packages for Raspberry Pi 4 (bcm2711-eeprom) and Raspberry Pi 5 (bcm2712-eeprom).

### CODE REVIEW

**Comment by mj22226** [comment]
Compile tested: bcm2711/2712 main
Run tested: RPi4/RPi5 main
```
root@OpenWrt:/#  rpi-eeprom-update
*** UPDATE AVAILABLE ***
BOOTLOADER: update available
   CURRENT: Wed Dec  6 18:29:25 UTC 2023 (1701887365)
    LATEST: Fri Jan  5 15:57:40 UTC 2024 (1704470260)
   RELEASE: stable (/lib/firmware/raspberrypi/bootloader-2712/stable)
            Use raspi-config to change the release.
root@OpenWrt:/# rpi-eeprom-update -a
*** PREPARING EEPROM UPDATES ***

BOOTLOADER: update available
   CURRENT: Wed Dec  6 18:29:25 UTC 2023 (1701887365)
    LATEST: Fri Jan  5 15:57:40 UTC 2024 (1704470260)
   RELEASE: stable (/lib/firmware/raspberrypi/bootloader-2712/stable)
            Use raspi-config to change the release.
   CURRENT: Wed Dec  6 18:29:25 UTC 2023 (1701887365)
    UPDATE: Fri Jan  5 15:57:40 UTC 2024 (1704470260)
    BOOTFS: /boot
Copying recovery.bin to /boot for EEPROM update

EEPROM updates pending. Please reboot to apply the update.
To cancel a pending update run "sudo rpi-eeprom-update -r".
root@OpenWrt:/# reboot
root@OpenWrt:/#  rpi-eeprom-update
BOOTLOADER: up to date
   CURRENT: Fri Jan  5 15:57:40 UTC 2024 (1704470260)
    LATEST: Fri Jan  5 15:57:40 UTC 2024 (1704470260)
   RELEASE: stable (/lib/firmware/raspberrypi/bootloader-2712/stable)
            Use raspi-config to change the release.
```
```
root@OpenWrt:/# rpi-eeprom-update
*** UPDATE AVAILABLE ***
BOOTLOADER: update available
   CURRENT: Tue Apr 26 10:24:28 UTC 2022 (1650968668)
    LATEST: Thu May 11 06:26:03 UTC 2023 (1683786363)
   RELEASE: stable (/lib/firmware/raspberrypi/bootloader-2711/stable)
            Use raspi-config to change the release.

  VL805_FW: Dedicated VL805 EEPROM
     VL805: up to date
   CURRENT: 000138c0
    LATEST: 000138c0
root@OpenWrt:/# rpi-eeprom-update -a
*** PREPARING EEPROM UPDATES ***

BOOTLOADER: update available
   CURRENT: Tue Apr 26 10:24:28 UTC 2022 (1650968668)
    LATEST: Thu May 11 06:26:03 UTC 2023 (1683786363)
   RELEASE: stable (/lib/firmware/raspberrypi/bootloader-2711/stable)
            Use raspi-config to change the release.

  VL805_FW: Dedicated VL805 EEPROM
     VL805: up to date
   CURRENT: 000138c0
    LATEST: 000138c0
   CURRENT: Tue Apr 26 10:24:28 UTC 2022 (1650968668)
    UPDATE: Thu May 11 06:26:03 UTC 2023 (1683786363)
    BOOTFS: /boot
Copying recovery.bin to /boot for EEPROM update

EEPROM updates pending. Please reboot to apply the update.
To cancel a pending update run "sudo rpi-eeprom-update -r".
root@OpenWrt:/# rpi-eeprom-update
BOOTLOADER: up to date
   CURRENT: Thu May 11 06:26:03 UTC 2023 (1683786363)
    LATEST: Thu May 11 06:26:03 UTC 2023 (1683786363)
   RELEASE: stable (/lib/firmware/raspberrypi/bootloader-2711/stable)
            Use raspi-config to change the release.

  VL805_FW: Dedicated VL805 EEPROM
     VL805: up to date
   CURRENT: 000138c0
    LATEST: 000138c0
```

**Comment by BKPepe** [comment]
This looks like casual update bump, but you added Git headers for patches, which are more changes than described in the commit, but also every time there is update to the latest version. Can we avoid to have some vague commit subjects like this and specify there exact version? I dont think there is anything wrong with that.

File: utils/bcm27xx-eeprom/Makefile [hunk 1]
```diff
@@ -45,15 +69,28 @@ define Package/bcm27xx-eeprom/install
...
-	$(INSTALL_DIR) $(1)/lib/firmware/raspberrypi/bootloader/stable
-	$(CP) $(PKG_BUILD_DIR)/firmware/stable/ $(1)/lib/firmware/raspberrypi/bootloader/
+	$(CP) $(PKG_BUILD_DIR)/firmware-2712/release-notes.md $(1)/lib/firmware/raspberrypi/bootloader-2712/
+	$(CP) $(PKG_BUILD_DIR)/firmware-2712/critical/ $(1)/lib/firmware/raspberrypi/bootloader-2712/
+	$(CP) $(PKG_BUILD_DIR)/firmware-2712/stable/ $(1)/lib/firmware/raspberrypi/bootloader-2712/
 endef
 
 $(eval $(call BuildPackage,bcm27xx-eeprom))
+$(eval $(call BuildPackage,bcm2711-eeprom))
+$(eval $(call BuildPackage,bcm2712-eeprom))
```
Comment (BKPepe) [inline utils/bcm27xx-eeprom/Makefile:96]: Maybe this should be done in separate commit.

Comment (Noltari) [reply utils/bcm27xx-eeprom/Makefile:96]: This is now a separate commit: b8b0e4549d80c0512084a9a2c82f280950d7ed74?w=1

**Comment by Noltari** [comment]
> This looks like casual update bump, but you added Git headers for patches, which are more changes than described in the commit, but also every time there is update to the latest version. Can we avoid to have some vague commit subjects like this and specify there exact version? I dont think there is anything wrong with that.

Sorry for that, should be fixed now.

**Comment by schuettecarsten** [comment]
The branch names `stable` and `critical` are outdated, `default` and `latest` are used now. It does not make sense to patch the original scripts to use the old names.

Also, script should not depend on the deprecated bcm27xx-userland package but on the new bcm27xx-utils package, if possible (did not check that myself)

**Comment by Noltari** [comment]
> The branch names `stable` and `critical` are outdated, `default` and `latest` are used now. It does not make sense to patch the original scripts to use the old names.

You're right, I will change the package to use the new names.

> Also, script should not depend on the deprecated bcm27xx-userland package but on the new bcm27xx-utils package, if possible (did not check that myself)

Since bcm27xx-utils provides bcm27xx-userland, you can try enabling bcm27xx-utils instead of bcm27xx-userland and you will still be able to select bcm27xx-eeprom.
In any case, bcm27xx-userland is still selected by default for bcm27xx, because I wanted bcm27xx-utils more testing before making it the default package.

**Comment by schuettecarsten** [comment]
```
root@GatewayDummi:~# rpi-eeprom-update -a
/usr/bin/rpi-eeprom-update: line 698: findmnt: not found
*** PREPARING EEPROM UPDATES ***
```

On my device, `rpi-eeprom-update` is complaining about missing `findmnt` command.

**Comment by Noltari** [comment]
> ```
> root@GatewayDummi:~# rpi-eeprom-update -a
> /usr/bin/rpi-eeprom-update: line 698: findmnt: not found
> *** PREPARING EEPROM UPDATES ***
> ```
> 
> On my device, `rpi-eeprom-update` is complaining about missing `findmnt` command.

Yes, I solved that on one of my latest forced pushes by adding `+mount-utils` to bcm27xx-eeprom.
If you install mount-utils with `opkg install mount-utils` that won't happen.
